### PR TITLE
DM-37823: Add toAstropy to DateTime

### PR DESCRIPTION
**Pull request by parejkoj**

### CODE REVIEW

File: python/lsst/daf/base/dateTime/dateTimeContinued.py [hunk 1]
```diff
@@ -19,11 +19,11 @@
 # You should have received a copy of the LSST License Statement and
 # the GNU General Public License along with this program.  If not,
 # see <http://www.lsstcorp.org/LegalNotices/>.
-#
-
 
 __all__ = []
 
+import astropy.time
```
Comment (timj) [inline python/lsst/daf/base/dateTime/dateTimeContinued.py:25]: I would be inclined to put the astropy.time import inside `toAstropy` because it's a very large package and nothing below daf_base is importing it. In the context of calling this from pipe_tasks or something that doesn't really matter (since astropy will be imported by daf_butler) but thinking of daf_base on its own it would be better, I think, to treat astropy as an optional dependency.

File: tests/test_dateTime.py [hunk 2]
```diff
@@ -30,10 +30,11 @@
 
 from lsst.daf.base import DateTime
 import lsst.pex.exceptions as pexExcept
+import lsst.utils.tests
 
 
-class DateTimeTestCase(unittest.TestCase):
-    """A test case for DateTime.
+class DateTimeTestCase(lsst.utils.tests.TestCase):
+    """Base class for tests of of DateTime objects, without any timezone.
```
Comment (timj) [inline tests/test_dateTime.py:37]: typo: of of

File: tests/test_dateTime.py [hunk 3]
```diff
@@ -395,6 +396,26 @@ def testPickle(self):
         nts = pickle.loads(pickle.dumps(ts))
         self.assertEqual(nts.nsecs(DateTime.UTC), int(1192755473000000000))
 
+    def testToAstropy(self):
+        """Test that DateTime converstion to astropy is exact to within machine
+        precision.
+        """
+        date = DateTime(0)
+        astropyTime = date.toAstropy()
+        self.assertFloatsAlmostEqual(date.get(), astropyTime.to_value(format="mjd"))
```
Comment (timj) [inline tests/test_dateTime.py:405]: Why is this needed instead of `self.assertAlmostEqual` ?  Aren't these all scalars?

**Review by timj — APPROVED** [review]
Looks good. A couple of minor questions.

File: tests/test_dateTime.py (hunk 3 above)
Comment (parejkoj) [reply tests/test_dateTime.py:405]: assertAlmostEqual uses a quite imprecise default and checks "decimal places" not "significant digits". Our `assertFloatsAlmostEqual` works for scalars as well, and defaults to machine precision.
